File: leetcode_c_plus_plus/26.cpp
// 26. 删除有序数组中的重复项
// https://leetcode-cn.com/problems/remove-duplicates-from-sorted-array/
using namespace std;
#include <vector>
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int length  = nums.size();
        if(length == 0){
            return 0;
        }
        int left = 0;
        int right = 0;
        while(right < length){
            if( right>0 &&  nums[left] != nums[right]){
                left++;
                nums[left] = nums[right];
            }
            right++;
        }
        return left+1;
    }
};

File: 剑指offer/python/26.py
# 剑指 Offer 26. 树的子结构
# https://leetcode-cn.com/problems/shu-de-zi-jie-gou-lcof/
# Definition for a binary tree node.
from typing import List
from collections import deque
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def isSubStructure(self, A: TreeNode, B: TreeNode) -> bool:
        if not B or not A:
            return False
        d = deque()
        d.append(A)
        
        def recur(A, B):
            if not B: return True
            if not A or A.val != B.val: return False
            return recur(A.left, B.left) and recur(A.right, B.right)

        while d:
            node = d.popleft()
            if node.val == B.val:
                if recur(node,B):
                    return True
            if node.left:
                d.append(node.left)
            if node.right:
                d.append(node.right)
        
        return False